doc/cephfs/mds-journaling.rst

```diff
@@ -0,0 +1,57 @@
+.. _mds-journaling:
+
+====================
+CephFS Metadata Pool
+====================
+
+CephFS uses a separate (metadata) pool for managing file metadata (inodes and
+dentries) in a Ceph File System. The metadata pool has all the information about
+files in a Ceph File System including the File System heirarchy. Additionally,
+CephFS maintains meta information related to other entities in a filesystem
+such as file system journals, open file table, session map, etc.
+
+This document describes how Ceph Metadata Server use and rely on journaling.
+
+=====================
+CephFS MDS Journaling
+=====================
+
+CephFS metadata servers stream a journal of metadata events into RADOS in the metadata
+pool prior to executing a file system operation. Active MDS daemon(s) manage metadata
+for files and directories in CephFS.
+
+CephFS uses journaling for couple of reasons:
+
+#. Consistency: On an MDS failover, the journal events can be replayed to reach a
+                consistent filesystem state. Also, metadata operations that require
+                multiple updates to the backing store need to be journaled for crash
+                consistency (along with other consistency mechanisms such as locking,
+                etc..).
+
+#. Performance: Journal updates are (mostly) sequential, hence updates to journals
+                are fast.
+
```

Comment:
A large journal also indirectly helps with getting the cache warm during failover.